crates/sui-framework/packages/move-stdlib/sources/ascii.move

```diff
@@ -67,18 +67,37 @@ module std::ascii {
         true
     }
 
+    /// Push a `Char` to the end of the `string`.
     public fun push_char(string: &mut String, char: Char) {
         string.bytes.push_back(char.byte);
     }
 
+    /// Pop a `Char` from the end of the `string`.
     public fun pop_char(string: &mut String): Char {
         Char { byte: string.bytes.pop_back() }
     }
 
+    /// Returns the length of the `string` in bytes.
     public fun length(string: &String): u64 {
         string.as_bytes().length()
     }
 
+    /// Append the `other` string to the end of `string`.
+    public fun append(string: &mut String, other: String) {
+        string.bytes.append(other.bytes)
+    }
+
+    /// Copy the slice of the `string` from `i` to `j` into a new `String`.
+    public fun sub_string(string: &String, mut i: u64, j: u64): String {
+        assert!(i <= string.length() && j <= string.length() && i <= j, EInvalidIndex);
```

Comment:
Testing `i <= string.length()` is redundant.

```suggestion
        assert!(i <= j && j <= string.length(), EInvalidIndex);
```

Reply:
do we include cases where `i == j` if while loop requires `i < j` ?

Reply:
`i == j` case is technically valid -- just means an empty slice.